src/main/java/subway/stations/SubwayStations.java

```diff
@@ -30,6 +30,61 @@ public int getObjectId() {
         }
     }
 
+    public List<Station> dijkstrasAlgorithm(Station source, Station destination, SubwayLines lines) {
+
+        Set<Station> unvisitedStations = new HashSet<>();
+
+        for(Station station : stations){
+            station.distanceFromSource = Integer.MAX_VALUE;
+            station.connections = getConnections(lines, station);
+            unvisitedStations.add(station);
+        }
+        Station currentSource = source;
+        currentSource.distanceFromSource = 0;
+        int shortestDistance;
+
+        while (!(currentSource.connections.contains(destination))) {
+            for(Station station : currentSource.connections){
+                if(unvisitedStations.contains(station)){
+                    station.distanceFromSource = currentSource.distanceFromSource + 1;
+                    station.previous = currentSource;
+                }
+            }
+            unvisitedStations.remove(currentSource);
+            shortestDistance = currentSource.distanceFromSource + 1;
+
+            for (Station station : unvisitedStations) {
```

Comment:
Why do you have two for loops, this one and the one on line 62?

Reply:
the shortest distance in the unvisited list is not always less than the current distance, so I first check for a distance that is equal to the current shortest distance, but then I have to do a second round of checking to see if there is in fact a shorter distance. I know it is overkill, but I was having trouble with the syntax for using a stream to do it instead

Reply:
Your goal is to get the station with the shortest distance out of the unvisited stations.
You should be able to do this with one for loop.